native/src/main/java/io/ballerina/lib/data/csvdata/csv/CsvParser.java

```diff
@@ -334,6 +391,181 @@ public Object execute(Reader reader, Type type, CsvConfig config, BTypedesc bTyp
+        }
+
+        private String buildJsonLog(Environment environment, Exception exception,
+                                    String sourceData, boolean excludeSourceData) {
+            String time = Instant.now().toString();
+            String message = exception.getMessage();
+            String json = "{\"time\":\"" + StringUtils.fromString(time) + "\"," +
+                    "\"location\":{\"row\":" + (this.lineNumber + 1) +
+                    ",\"column\":" + (this.columnIndex + 1) + "}," +
+                    (excludeSourceData ? "" : "\"sourceData\":\"" + StringUtils.fromString(sourceData.trim()) + "\",") +
+                    "\"message\":\"" + StringUtils.fromString(message) + "\"" +
+                    "}";
+            StrandMetadata strandMetadata = new StrandMetadata(true,
+                    ModuleUtils.getProperties(TO_JSON_METHOD));
+            Object[] arguments = new Object[]{StringUtils.fromString(json)};
+            Object result = environment.getRuntime().callFunction(ModuleUtils.getModule(), TO_JSON_METHOD,
+                    strandMetadata, arguments);
+            return result.toString();
+        }
+
+        private void handleFileOutputLogging(Environment environment, BMap<?, ?> logFileConfig, Exception exception,
+                                             AtomicBoolean isOverwritten, String rawData, String dataType) {
+            if (logFileConfig == null || !logFileConfig.containsKey(FILE_PATH)) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_CONFIGURATIONS, MISSING_FILE_PATH_ERROR);
+            }
+            BString filePathValue = logFileConfig.getStringValue(FILE_PATH);
+            String filePath = filePathValue != null ? filePathValue.toString() : null;
+            if (filePath == null || filePath.isBlank()) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_CONFIGURATIONS, MISSING_FILE_PATH_ERROR);
+            }
+            handleLogFileGeneration(filePath);
+            String fileWriteOption = logFileConfig.getStringValue(FILE_WRITE_OPTION).toString();
+            if (OVERWRITE.equals(fileWriteOption) && !isOverwritten.get()) {
+                overwriteLogFile(filePath);
+                isOverwritten.set(true);
+            }
+            if (dataType.equals(RAW)) {
+                writeLogsToFile(filePath, rawData + "\n");
+            } else {
+                boolean excludeSourceData = dataType.equals(METADATA);
+                String jsonLog = buildJsonLog(environment, exception, rawData, excludeSourceData);
+                writeLogsToFile(filePath, jsonLog + "\n");
+            }
+        }
+
+        private boolean isAllowedFailSafe(Exception exception) {
+            String message = exception.getMessage();
+            if (message == null) {
+                return true;
+            }
+            String normalized = message.toLowerCase();
+            return !normalized.contains("invalid csv data format")
```

Comment:
Don't we have specific exception types?

Reply:
Fixed in [8f40012](https://github.com/ballerina-platform/module-ballerina-data.csv/pull/60/commits/8f40012a136fbe852e93a07d8f522b05efc792aa)